src/sentry/integrations/slack/notify_action.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from sentry import http
+from sentry.rules.actions.base import EventAction
+from sentry.utils import metrics, json
+from sentry.models import Integration
+
+from .utils import build_attachment
+
+
+class SlackNotifyServiceForm(forms.Form):
+    team = forms.ChoiceField(choices=(), widget=forms.Select(
+        attrs={'style': 'width:150px'},
+    ))
+    channel = forms.CharField(widget=forms.TextInput(
+        attrs={'placeholder': 'i.e #critical-errors'},
+    ))
+
+    def __init__(self, *args, **kwargs):
+        team_list = [(i.id, i.name) for i in kwargs.pop('integrations')]
+        self.channel_transformer = kwargs.pop('channel_transformer')
+
+        super(SlackNotifyServiceForm, self).__init__(*args, **kwargs)
+
+        self.fields['team'].choices = team_list
+        self.fields['team'].widget.choices = self.fields['team'].choices
+
+    def clean_channel(self):
+        channel = self.cleaned_data.get('channel', '').lstrip('#')
+        team = self.cleaned_data.get('team')
+
+        channel_id = self.channel_transformer(team, channel)
+
+        if channel_id is None and team is not None:
+            params = {
+                'channel': channel,
+                'team': dict(self.fields['team'].choices).get(int(team)),
+            }
+
+            raise forms.ValidationError(
+                _('The #%(channel)s channel does not exist in the %(team)s Slack team.'),
+                code='invalid',
+                params=params,
+            )
+
+        return channel
+
+
+class SlackNotifyServiceAction(EventAction):
+    form_cls = SlackNotifyServiceForm
+    label = 'Send a notification to the Slack {team} team in {channel}'
```

Comment:
This should probably be unicode. I'm not sure what Slack's rules are on names here, but probably safer to just use unicode.

Reply:
caab69a5a063e54aa1fadb572921f80e8c9f297a it does have `format` called so just to be safe.

https://github.com/getsentry/sentry/blob/3a9695bd223082de497b725aa62b4571f08b5e8d/src/sentry/rules/base.py#L86